src/core/validation/validators/MaxAmountValidator.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 NEM (https://nem.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+import { Validator, staticImplements } from './Validator';
+
+@staticImplements<Validator>()
+export class MaxAmountValidator {
+    /**
+     * Validates the max amount of the mosaic supply. eg: 9,000,000,000
+     * @static
+     * @param {*} relativeAmount
+     * @returns {boolean}
+     */
+    public static validate(relativeAmount: string): boolean {
+        return Number(relativeAmount) <= 9_000_000_000;
```

Comment:
I think `9_000_000_000` should not be hardcoded.
Supply may differ depends on mosaic

Reply:
yeah, base on my understanding the validation rules only take care of the [standard mosaic validation](https://docs.symbolplatform.com/concepts/mosaic.html#initial-supply).

The rest will pass to the server for verification. @bassemmagdy correct me if I'm wrong.

But I will revisit the issue again.

Reply:
@AnthonyLaw it shouldn't be hard coded IMO, furthermroe instead of limiting validation to `9000_000_000` which is not the max allowed isn't fixing the [issue](https://github.com/nemgrouplimited/symbol-desktop-wallet/issues/758) as the validation should be `<=9_000_000_000_000_000`, @cryptoBeliever what do you think?

Reply:
8_999_999_999_000_000 is max value for mosaic creation.

Reply:
I will revisit this task.

Reply:
@bassemmagdy The reason used `9_000_000_000` for amount field is because relative amount.

Reply:
Hmm. Not sure if I understand. Max releative amount for coin with divisibility=0 is 8_999_999_999_000_000:

http://explorer.testnet.symboldev.network/mosaics/5D83F9AE97B99686